httpclient5/src/main/java/org/apache/hc/client5/http/cookie/Cookie.java

```diff
@@ -77,9 +78,26 @@
      * considered immutable. Changing it (e.g. using setTime()) could result
      * in undefined behaviour. Do so at your peril. </p>
      * @return Expiration {@link Date}, or {@code null}.
+     * @deprecated Use {{@link #getExpiryInstant()}}
      */
+    @SuppressWarnings("DeprecatedIsStillUsed")
+    @Deprecated
     Date getExpiryDate();
 
+    /**
+     * Returns the expiration {@link Instant} of the cookie, or {@code null} if none exists.
+     * <p><strong>Note:</strong> the object returned by this method is
+     * considered immutable. Changing it (e.g. using setTime()) could result in undefined behaviour.
+     * Do so at your peril. </p>
+     *
+     * @return Expiration {@link Instant}, or {@code null}.
+     * @since 5.2
+     */
+    default Instant getExpiryInstant() {
+        final Date date = getExpiryDate();
```

Comment:
@arturobernalg This is where you should be suppressing deprecation warnings.

Reply:
idem above. but I can changed.  Please let me what do you prefer.